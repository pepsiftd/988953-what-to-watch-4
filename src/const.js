import PropTypes from 'prop-types';

const MAX_GENRES_AMOUNT = 9;
const MAX_CARDS_ON_MOVIE_PAGE = 4;
const ALL_GENRES_FILTER = `All genres`;
const MAX_ACTORS_IN_SHORT_DETAILS = 4;
const API_REQUEST_TIMEOUT_IN_MS = 1000 * 5;
const COMMENT_MIN_LENGTH = 50;
const COMMENT_MAX_LENGTH = 400;

const Url = {
  BASE: `https://4.react.pages.academy/wtw`,
  FILES_STORAGE: `https://4.react.pages.academy`,
};

const MovieInfoTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const VIDEO_PLAYER_SETTINGS = {
  isMute: false,
  autoplay: true,
  width: `100%`,
  height: `100%`,
};

const SMALL_CARD_VIDEO_SETTINGS = {
  isMute: true,
  width: 280,
  height: 175,
};

const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  MOVIE_PAGE: `/films`,
  PLAYER: `/player`,
};

const RequestRoute = {
  LOGIN: `/login`,
  FILMS: `/films`,
  FAVORITE: `/favorite`,
  PROMO: `/films/promo`,
  COMMENTS: `/comments`,
};

const RATING_LEVEL_MAP = {
  0: `Bad`,
  3: `Normal`,
  5: `Good`,
  8: `Very good`,
  10: `Awesome`,
};

const FILM_OBJECT_PROP_TYPES = {
  id: PropTypes.number.isRequired,

  title: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  imageSrc: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  fullVideo: PropTypes.string.isRequired,

  poster: PropTypes.string.isRequired,
  backgroundImage: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  scoresCount: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  runTime: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
};

export {
  AppRoute,
  RATING_LEVEL_MAP,
  FILM_OBJECT_PROP_TYPES,
  VIDEO_PLAYER_SETTINGS,
  Url,
  MAX_GENRES_AMOUNT,
  ALL_GENRES_FILTER,
  MAX_CARDS_ON_MOVIE_PAGE,
  COMMENT_MIN_LENGTH,
  COMMENT_MAX_LENGTH,
  MovieInfoTab,
  SMALL_CARD_VIDEO_SETTINGS,
  MAX_ACTORS_IN_SHORT_DETAILS,
  API_REQUEST_TIMEOUT_IN_MS,
  RequestRoute,
};
