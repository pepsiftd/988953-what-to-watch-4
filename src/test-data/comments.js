const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Courtney Cox`,
    },
    rating: 5.1,
    comment: `I don't know what to say but I can't help posting`,
    date: new Date(`August 3, 2020 UTC`),
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `Mockingbird`,
    },
    rating: 8.3,
    comment: `This is not a movie of a simple kind`,
    date: new Date(`July 15, 2020 UTC`),
  },
  {
    id: 3,
    user: {
      id: 154,
      name: `Pelagia`,
    },
    rating: 1.5,
    comment: `Super mega exquisitely boring`,
    date: new Date(`June 1, 2020 UTC`),
  }
];

export {reviews};
