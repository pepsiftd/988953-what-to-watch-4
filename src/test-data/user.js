const userInfo = {
  id: 0,
  email: `example@email.com`,
  name: `Sarah Connor`,
  avatar: `img/avatar.jpg`,
};

export {userInfo};
