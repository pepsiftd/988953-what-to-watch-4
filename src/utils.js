const getUniqueItems = (list) => {
  return list.filter((it, i, array) => {
    return array.indexOf(it) === i;
  });
};

const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getMoviesByGenre = (movies, genre) => {
  return genre === `All genres` ? movies : movies.filter((movie) => movie.genre === genre);
};

const getGenresFromMovies = (movies) => {
  return getUniqueItems(movies.map((movie) => movie.genre));
};

export {extend, getMoviesByGenre, getUniqueItems, getGenresFromMovies};
