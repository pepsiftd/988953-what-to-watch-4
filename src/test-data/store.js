import {NameSpace} from '@/reducer/name-space';
import {movies, promoMovie} from '@/test-data/movies';
import {userInfo} from '@/test-data/user';
import {reviews} from '@/test-data/comments';
import {ALL_GENRES_FILTER} from '@/const';

const store = {
  [NameSpace.DATA]: {
    movies,
    promoMovie,
    favoriteMovies: [],
    reviews,
  },
  [NameSpace.APP]: {
    currentGenre: ALL_GENRES_FILTER,
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTHORIZED`,
    authorizationError: 200,
    authorizationInfo: userInfo,
  },
};

export {store};
