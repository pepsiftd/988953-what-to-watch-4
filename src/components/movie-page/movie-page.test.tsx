import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {history} from '@/history';
import {noop} from '@/utils';
import {MovieInfoTab} from '@/const';

import MoviePage from './movie-page';
import {store as testStoreData} from '@/test-data/store';

const mockStore = configureStore([]);

it(`MoviePage should render correctly`, () => {
  const store = mockStore(testStoreData);

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <MoviePage
            id={1}
            setActiveItem={noop}
            activeItemId={MovieInfoTab.OVERVIEW}
          />
        </Provider>
      </Router>,
      {
        createNodeMock() {
          return {};
        }
      }
  );

  expect(tree).toMatchSnapshot();
});
