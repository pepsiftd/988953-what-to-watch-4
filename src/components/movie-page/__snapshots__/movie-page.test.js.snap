// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="I, Robot"
          src="backgroundImage.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <a
              href="/mylist"
              onClick={[Function]}
            >
              <img
                alt="User avatar"
                height="63"
                src="/img/avatar.jpg"
                width="63"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            I, Robot
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              drama
            </span>
            <span
              className="movie-card__year"
            >
              1994
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="I, Robot poster"
            height="327"
            src="bigimage.jpgg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              155,0
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                
              </span>
              <span
                className="movie-rating__count"
              >
                5
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            NICE
            <p
              className="movie-card__director"
            >
              <strong>
                Director: John connor
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: chubakka, leia, skywalker
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
